command/server.go

```diff
@@ -1692,6 +1690,23 @@ func (c *ServerCommand) setupTelemetry(config *server.Config) (*metrics.InmemSin
 
 	// Configure the statsite sink
 	var fanout metrics.FanoutSink
+
+	// Configure the Prometheus sink
+	if telConfig.PrometheusRetentionTime == 0 {
+		return nil, fmt.Errorf("telemetry.prometheus_retention_time must be > 0")
+	}
+
+	prometheusOpts := prometheus.PrometheusOpts{
+		Expiration: telConfig.PrometheusRetentionTime,
+	}
+
+	sink, err := prometheus.NewPrometheusSinkFrom(prometheusOpts)
```

Comment:
Wouldn't you only want to do this if the retention time is not zero? That way it'd gate it based on configuration existing, like the other types.

Reply:
Sorry, I may have misunderstood your previous comment then :(
You mentioned it might be better if users have all formats available with no gates, so right now, specifying a 0 value for Prometheus retention is return an error, with the default being 24h retention

Would you prefer that the behavior is still disabled if an explicit 0 is used ?